internal/pkg/model/naming.go

```diff
@@ -1,459 +0,0 @@
-package model
-
-import (
-	"fmt"
-	"sync"
-
-	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
-	"github.com/keboola/keboola-as-code/internal/pkg/utils"
-)
-
-const (
-	MetaFile          = "meta.json"
-	ConfigFile        = "config.json"
-	DescriptionFile   = "description.md"
-	PhaseFile         = "phase.json"
-	TaskFile          = "task.json"
-	CodeFileName      = `code` // transformation code block name without ext
-	blocksDir         = `blocks`
-	blockNameTemplate = utils.PathTemplate(`{block_order}-{block_name}`)
-	codeNameTemplate  = utils.PathTemplate(`{code_order}-{code_name}`)
-	phasesDir         = `phases`
-	phaseNameTemplate = utils.PathTemplate(`{phase_order}-{phase_name}`)
-	taskNameTemplate  = utils.PathTemplate(`{task_order}-{task_name}`)
-	SqlExt            = `sql`
-	PyExt             = `py`
-	JuliaExt          = `jl`
-	RExt              = `r`
-	TxtExt            = `txt`
-	SqlComment        = `--`
-	PyComment         = `#`
-	JuliaComment      = `#`
-	RComment          = `#`
-	TxtComment        = `//`
-)
-
-// Naming of the files.
-type Naming struct {
```

Comment:
1. commit

`modelNaming` robil to privela a musel som ho rozdelit na:
- `Template` ... format generovanych path, ... vsetky fields nizsie, okrem `used*`.
- `Registry` - zoznam existujucich `paths`, moznost ziskat `key -> path`, `path -> key`, ... `used*` fields
- `Matcher` - pouziva sa pri `persist`, ... umoznuje z path zistit, ci sa jedna o config, row ...
- `Generator` - generuje `paths` pre nove objekty